tactics/tactics.ml

```diff
@@ -2796,7 +2796,22 @@ let generalize_goal_gen env sigma ids i ((occs,c,b),na) t cl =
   let open Context.Rel.Declaration in
   let decls,cl = decompose_prod_n_assum sigma i cl in
   let dummy_prod = it_mkProd_or_LetIn mkProp decls in
-  let newdecls,_ = decompose_prod_n_assum sigma i (subst_term_gen sigma EConstr.eq_constr_nounivs c dummy_prod) in
```

Comment:
Why is this comparing with nounivs? Could you try with the regular replace_term and see impact on CI?

Reply:
It's probably here for a reason, and this would be a change unrelated to this PR which is 1:1 semantically.

Reply:
To be more precise, I don't mind trying to tweak this but this needs to go somewhere else. Btw this change is here since the very commit that introduced universe polymorphism.

Reply:
After reflection, I think I know why this is done this way. The `generalize` tactic intuitively takes a pattern as argument, but it's actually a constr whose universes have been autogenerated. So we have to rely on this trick to make examples containing types to work.

Reply:
I'd like a comment to attract the reader's attention to the _nounivs, as the rest of the boilerplate distracted me and made me miss it initially.
Then I'll merge.

Reply:
Done.